src/hooks/usePairTokenPriceByTimestamp.ts

```diff
@@ -0,0 +1,52 @@
+import { Currency, Pair, Token } from '@swapr/sdk'
+
+import { useEffect, useState } from 'react'
+
+import {
+  convertToChartData,
+  GetBlockPairTokenPriceQueryData,
+  TIMEFRAME_PROPRETIES,
+} from '../components/Charts/chartUtils'
+import { useGetPairTokenPricesQuery } from '../graphql/generated/schema'
+import { wrappedCurrency } from '../utils/wrappedCurrency'
+
+import { useActiveWeb3React } from '.'
+
+type PairTokenPriceByTimestampProps = {
+  currency0?: Currency
+  currency1?: Currency
+  dateInterval: string
+}
+
+export function usePairTokenPriceByTimestamp({ currency0, currency1, dateInterval }: PairTokenPriceByTimestampProps) {
+  const { chainId } = useActiveWeb3React()
+  const [wrappedToken1, setWrappedToken1] = useState<Token>()
+  const [pairAddress, setPairAddress] = useState<string>()
+
+  const { data, loading, error } = useGetPairTokenPricesQuery({
+    variables: {
+      pairAddress,
+      timestamp: TIMEFRAME_PROPRETIES[dateInterval].timestamp,
+      timeframe: TIMEFRAME_PROPRETIES[dateInterval].pairTokenPriceTimeframe,
+    },
+  })
+
+  useEffect(() => {
+    try {
```

Comment:
do we need `try catch` here?

Reply:
@berteotti what do you think?

Reply:
This will throw an exception if we call this on a chain without liquidity (e.g. Polygon, Optimism). I think this is the best solution to this problem, we don't have to change anything once there is liquidity in this chains.